lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2309,20 +2310,46 @@ namespace {
         }
       }
 
-      // Check that we have a distributed function.
-      auto func = dyn_cast<AbstractFunctionDecl>(decl);
-      if (!func || !func->isDistributed()) {
-        ctx.Diags.diagnose(declLoc,
-                           diag::distributed_actor_isolated_method)
-          .fixItInsert(decl->getAttributeInsertionLoc(true), "distributed ");
+      // Cannot reference subscripts, or stored properties.
+      auto var = dyn_cast<VarDecl>(decl);
+      if (isa<SubscriptDecl>(decl) || var) {
```

Comment:
The rewritten logic here silently accepts declaration kinds it doesn't understand. I'm not sure if I can trigger a bug based on this, but the prior formulation (`!func || !func->isDistributed()`) was more robust. Can we use more straightforward logic here like:

```
if (auto func = dyn_cast<AbstractFunctionDecl>(decl)) {
 handle funcs and return
}

if (auto var = dyn_cast<VarDecl>(decl)) {
  handle vars and return
}

diagnose and return None
```